src/parser/scanner_support.c

```diff
@@ -75,6 +87,13 @@ struct st_hint_msg
     m_used = 0;
     msg_ptr = msg_buf;
     msg_buf[0] = 0x00;
+
+    m_hint_ptr = NULL;
+    memset (&m_hint_head, 0x00, sizeof (STMT_HINT));
+    m_hint_head.stmt_no = -1;
```

Comment:
How about implementing the default constructor for STMT_HINT?

Reply:
I don't think there's any need to introduce a constructor as it's a simple code.